packages/eslint-plugin-gestalt/src/no-spread-props.js

```diff
@@ -0,0 +1,95 @@
+// @flow strict
+import {
+  buildPropsFromKeyValuesVariable,
+  hasImport,
+  getComponentNameFromAttribute,
+  getNamedImportsComponents,
+  getPropertiesFromVariable,
+  getVariableNodeInScopeFromName,
+} from './eslintASTHelpers.js';
+/**
+ * @fileoverview Prevent spreading props in Gestalt components to enable AST codemods and usage-metrics scripts.
+ * @author Alberto Carreras <acarreras@pinterest.com>
+ */
+
+const rule = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Prevent spreading props in Gestalt components to enable AST codemods and usage-metrics scripts.',
+      category: 'Gestalt alternatives',
+      recommended: true,
+      url: 'https://gestalt.pinterest.systems/Eslint%20Plugin#gestaltno-spread-props',
+    },
+    fixable: 'code',
+    schema: ([]: $ReadOnlyArray<empty>),
+    messages: {
+      disallowed: `Prop spreading in Gestalt component '{{ name }}' is forbidden, write your props out instead. {{ autofix }}`,
+    },
+  },
+
+  // $FlowFixMe[unclear-type]
+  create(context: Object): Object {
```

Comment:
Wow. I love how readable this code is now with the helpers! Far fewer comments needed, so much easier to follow. This is awesome!